scripts/core/EXP-COMMAND-BUFFER.rst

```diff
@@ -138,14 +140,60 @@ were obtained from.
+                                        nullptr, 0, nullptr, &syncPoint, nullptr,
+                                        nullptr);
 
     // Append a kernel launch with syncPoint as a dependency, ignore returned
     // sync-point
     ${x}CommandBufferAppendKernelLaunchExp(hCommandBuffer, hKernel, workDim,
                                            pGlobalWorkOffset, pGlobalWorkSize,
                                            pLocalWorkSize, 1, &syncPoint,
-                                           nullptr, nullptr);
+                                           nullptr, 0, nullptr, nullptr,
+                                           nullptr);
+
+Command Synchronization With Events
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+When appending commands to a command-buffer an optional ``phEventWaitList``
+input parameter is available for passing a list of ${x}_event_handle_t objects
+the command should wait on. As well as an optional ``phEvent`` output parameter
+to get a ${x}_event_handle_t object that will be signaled on completion of the
+command execution. It is the users responsibility to release the returned
+``phEvent`` with ${x}EventRelease.
+
+This returned signal event is only valid for the last execution of the command,
+and prior to the first execution of the command-buffer it represents an empty
+submission that is considered complete. When a user calls
+${x}CommandBufferEnqueueExp all the signal events returned from the individual
+commands in the command-buffer are synchronously reset to a non-complete state
+prior to the asynchronous commands beginning.
+
+The wait event parameter allows commands in a command-buffer to depend on the
+completion of UR commands submitted to a queue which are external
+to a command-buffer. While the output signal event parameter allows individual
+commands in a command-buffer to trigger external queue commands. Using returned
+signal events as wait events inside the same command-buffer is also valid usage.
+
+It is possible for commands in different command-buffer objects to synchronize
+using the event mechanism. This is only guaranteed to behave correctly in the one
+directional synchronization case, where the signal events of one
+command-buffer's commands are used as a wait events of another command-buffer's
+commands. Such a relationship defines a permanent dependency between the
+command-buffers which does not need to be updated using
+:ref:`command event update` to preserve synchronization on future enqueues of
+the command-buffer.
+
+Bi-directional sync between individual commands in two separate command-buffers
+is however not guaranteed to behave correctly. This is due to the completion
+state of the command events only being reset when a command-buffer is executed.
+It is therefore possible for the first command-buffer enqueued to execute the
+wait node that needs to be reset by the enqueue of the second command-buffer,
```

Comment:
In general I think this wording is good but I'd probably clarify this sentence a little, it reads like the wait in the first command-buffer is not guaranteed to be reset but I think it's more accurate to say that the event the wait node is waiting on is not guaranteed to be reset.